extensions/vscode/.eslintrc.json

```diff
@@ -25,8 +25,14 @@
     "@typescript-eslint/semi": "warn",
     "curly": "warn",
     "eqeqeq": "warn",
-    "no-throw-literal": "warn",
-    "semi": "off"
+    "no-throw-literal": "error",
+    "semi": "off",
+    "no-restricted-syntax": [
+      "error",
+      {
+        "selector": "ThrowStatement"
+      }
+    ]
```

Comment:
I'm not in love with the pattern of restricting `throw` entirely because I don't personally like returning `undefined` if there is an error.

It doesn't feel like the _normal_ JavaScript pattern to me. Having said that though this is a great way to keep us from just throwing without actually showing an error message.

I don't have a good alternative to allow both my preference and the benefit we are getting from this, but just voicing this for potential future changes to our pattern.